### DISCUSSION TITLE
No new features (yet), "just" language updates

### DISCUSSION THREAD

**Discussion by NiyaShy** [NiyaShy/XB1ControllerBatteryIndicator#69] [Releases]
\*"last unicorn" plays in the background\* "I'm aliiiiiiive~"
Jokes aside, this isn't really a new version but "just" a new release to show the project is still active (since the last proper release was from 2019 with "silent" updates for new languages).

To mention it here as well without posting a full release, you can try out an unfinished version of what will most likely become 1.4 [here](https://github.com/NiyaShy/XB1ControllerBatteryIndicator/discussions/55). It contains new features (outlined in the post) including a new overlay, but most of the translations haven't been updated yet and there's still some fine-tuning to be done. 

And we also reached another "milestone" concerning downloads: the last version I uploaded May 2021 has so far been downloaded **over 20.000 times!** Really happy that so many people find this little project useful.

<hr /><em>This discussion was created from the release <a href='https://github.com/NiyaShy/XB1ControllerBatteryIndicator/releases/tag/v1.3.1.1'>No new features (yet), "just" language updates</a>.</em>

**Comment by manuelgrabowski** [comment]
Looking forward to the update – do you have a Patreon or any other avenues for donations? Would love to give back a small token of appreciation for this amazing little tool that I've been using for years now. :)

**Reply by NiyaShy** [reply]
Hey, thanks for asking, but I'm not planning to accept any form of donations, mainly because most of the new features came from other people and I'd feel kind of guilty for reaping the rewards of their hard work. But I really appreciate that you like it that much!

Biggest reward I'm already getting is that so many use it and spread the word about its existence :)

**Comment by Ryonez** [comment]
Just a heads up the version numbers for the exe aren't matching the release version.

![image](https://user-images.githubusercontent.com/16157130/206893799-5ccb56b8-bc77-4ef8-b21d-35035a6a36b7.png)

**Reply by NiyaShy** [reply]
You're right, I didn't change it in the code. For one because I simply forgot about it 😅, and because the increment on the 4th digit was more or less just "cosmetic" so it shows up as a new release. Actually using the 4th digit to show that something minor like a new language was changed/added is a thing I was planning to do from 1.4 onwards (and adjusting the update check so you can chose if you want to get notified about any new version or just "bigger" updates).

**Comment by liorth** [comment]
![image](https://user-images.githubusercontent.com/30504031/231532660-12b4b1bf-1168-4c11-a350-8f82f9234db8.png)
It had worked for my old xbox one controller. But since I bought a new one which is the new model, it stays like this.

**Reply by NiyaShy** [reply]
Hey,
this has sometimes happened ever since I started on this project, and I still don't know why the controllers sometimes don't report anything right away. Can sadly only give you the "usual" advice: use the controller for a while and see if it changes.

Asking just in case: both your old and new controller were/are connected via the proprietary dongle and not via Bluetooth, right?

**Reply by liorth** [reply]
I have used the new controller for like months, still no changing.

For your asking: it is actually the opposite, both controllers were/are connect via Bluetooth. I don't even see a dongle in the package. If I recall it correctly, since 8 Jahres ago when I bought my first xbox wireless controller for PC, there hasn't been a dongle anymore.

**Reply by liorth** [reply]
And your reply reminds me, that I could do a experiment. I connected the two controllers the same time. While the program shows me the battery status of my old controller, it still has no info for the new one. Another thing I have noticed while experimenting: windows 11 shows the battery of the new model in bluetooth settings. This new to me, maybe it has something to do with the bug? Even better, maybe just read the status of the system to use for the program? Windows 11 still doesn't have any notification for low battery.

**Reply by NiyaShy** [reply]
AFAIK the controllers were never offered with the dongle since you only need it when using them on windows, that's why it was always an optional/separate piece to buy. 

Had to look it up on [wikipedia](https://en.wikipedia.org/wiki/Xbox_Wireless_Controller), the first controller revision that supported BT was model 1708, released in 2016. So far I got reports that those sometimes, but not always, work and report data when connected via BT. But the newest revision (Series X/S, v1914, sold since 2020) has a different BT implementation that reports no or only garbage values through BT. See issue #49 for a lengthy discussion about it.

tl;dr of the thread: I can't do nothing about it, only way you'll get a battery level report out of that controller is via the proprietary dongle, not via BT.

**Reply by liorth** [reply]
I see. Well then I would consider to buy a adapter for that, if you are quite sure that it would work with the separate adapter. As it is helpful to know the battery is low when playing some action games. Besides it is not that expensive.

**Reply by NiyaShy** [reply]
Let me put it this way: I've bought 3 XBOne controllers over the years (model 1537, 1697 and 1914), and all of them have worked flawlessly with the (slightly oversized) first revision of the dongle that I got together with the first controller. Countless others gave feedback that their controllers work with the dongle (both old/big and newer/smaller revision), and a few reported that cheaper 3rd party dongles from china also worked for them. Only major detection/reporting issues came from users that tried BT with the newer controller models.

I can't say that reporting will work right away 100% of the time (the tool sometimes has its quirks), but your chances that it will work at all are very high.
